File: backend/app/Models/Student.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Student extends Model
{
    use HasFactory;
    use SoftDeletes;

    protected $fillable = [
        'user_id',
        'dorm_id',
        'full_name',
        'student_no',
        'phone',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function dorm()
    {
        return $this->belongsTo(Dorm::class);
    }

    public function roomAssignments()
    {
        return $this->hasMany(RoomAssignment::class);
    }

    public function activeAssignment()
    {
        return $this->hasOne(RoomAssignment::class)->where('active', true);
    }
}

File: mobile/lib/src/models/student.dart
class Student {
  Student({
    required this.id,
    required this.userId,
    required this.dormId,
    required this.fullName,
    required this.studentNo,
    this.phone,
    this.email,
  });

  final int id;
  final int userId;
  final int dormId;
  final String fullName;
  final String studentNo;
  final String? phone;
  final String? email;

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      dormId: json['dorm_id'] as int,
      fullName: json['full_name'] as String? ?? '',
      studentNo: json['student_no'] as String? ?? '',
      phone: json['phone'] as String?,
      email: json['email'] as String?,
    );
  }
}
